### PR TITLE
Adding in Github Actions for CI.  

### PR DESCRIPTION
**Pull request by trankin**
Linux seems fine, Windows was failing on node_js versions 8.x, 10.x and 11.x.  I've removed those in this pull request as I'm not sure if the problem is on the github side or the build code in the repository.

### CODE REVIEW

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: Linux Builds
+
+on: [push]
```
Comment (robertsLando) [inline .github/workflows/linux.yml:3]: could you run this even on pr please?

File: .github/workflows/linux.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        node-version: [0.12, 4.x, 6.x, 8.x, 10.x, 11.x, 12.x]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Install ubuntu-toolchain-r/test      
+      run: sudo add-apt-repository ppa:ubuntu-toolchain-r/test
+    - name: Install libudev-dev
+      run: sudo apt install libudev-dev
+    - name: Build open-zwave-latest
+      run: wget https://github.com/OpenZWave/open-zwave/archive/master.zip && unzip master.zip && cd open-zwave-master && sudo make install && sudo ldconfig /usr/local/lib /usr/local/lib64 && cd ..
```
Comment (robertsLando) [inline .github/workflows/linux.yml:21]: Please split this in multi-lines run for better readability

File: .github/workflows/linux.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+    - run: npm install -g node-gyp
```
Comment (robertsLando) [inline .github/workflows/linux.yml:26]: You can use: 

```yml
 run: |
          action \
          action2 \
          action n
```

File: .github/workflows/windows.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+      matrix:
+        node-version: [12.x]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+    - run: npm install -g node-gyp
```
Comment (robertsLando) [inline .github/workflows/windows.yml:20]: same here

**Comment by robertsLando** [comment]
windows build is failing because you haven't installed OZW

File: .github/workflows/linux.yml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        sudo make install 
+        sudo ldconfig /usr/local/lib /usr/local/lib64 
+        cd ..
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+    - run: npm install -g node-gyp
+    - run: npm install
+    - run: npm run build --if-present
```
Comment (robertsLando) [inline .github/workflows/linux.yml:34]: this is useless

File: .github/workflows/windows.yml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    - run: npm install
+    - run: npm run build --if-present
```
Comment (robertsLando) [inline .github/workflows/windows.yml:22]: same

File: .github/workflows/windows.yml [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    - run: npm test
+      env:
+        CI: true
```
Comment (robertsLando) [inline .github/workflows/windows.yml:25]: same

**Comment by trankin** [comment]
> windows build is failing because you haven't installed OZW

I believe part of the windows node-gyp build is to is to build and install OZW correct?  It's working on node 12.

**Comment by robertsLando** [comment]
> I  believe part of the windows node-gyp build is to is to build and install OZW correct

You are right (never used it under windows but by checking the code it seems to do it automatically)

So I think something fails in the install process, can you show me the error?

**Comment by trankin** [comment]
> > I  believe part of the windows node-gyp build is to is to build and install OZW correct
> 
> You are right (never used it under windows but by checking the code it seems to do it automatically)
> 
> So I think something fails in the install process, can you show me the error?

Are you able to see this? https://github.com/trankin/node-openzwave-shared/runs/750522182?check_suite_focus=true

**Comment by robertsLando** [comment]
@trankin 

> Error: spawn C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\15.0\Bin\MSBuild.exe ENOENT

**Comment by trankin** [comment]
Likely related to this: https://github.com/actions/virtual-environments/issues/68  We may need VS2017 for older versions of nodejs.  I won't be able to dig in and review this further for a few days, and can't guarantee that I'll come up with a solve in short order (not to say that it's a hard problem to solve, I have some digging to do as this I'm still getting familiar with github actions), you might consider merging this pull request and leave appveyor running until it's resolved to get older versions running in github actions, or I can remove the windows side of the build in github actions altogether until it's resolved.  Let me know how you'd like to see this move forward.

**Comment by robertsLando** [comment]
@trankin Don't worry thanks for your collaboration I will dig into this next days

**Pull request merged**